### PR TITLE
Update offer_id type to varchar

### PR DESCRIPTION
**Pull request by macdiesel**
Related Changes:

The reporting change PR:
 https://github.com/edx/warehouse-transforms/pull/4133/files

Prefect PR:
https://github.com/edx/prefect-flows/pull/199

Edx-analytics-api PR:
TODO
Version: 4.6.5

**Merge checklist:**
- [X] `./manage.py makemigrations` has been run

**Post merge:**
- [x] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/openedx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_data/__init__.py [hunk 1]
```diff
@@ -2,6 +2,6 @@
 Enterprise data api application. This Django app exposes API endpoints used by enterprises.
 """
 
-__version__ = "4.6.4"
+__version__ = "4.6.6"
```
Comment (muhammad-ammar) [inline enterprise_data/__init__.py:5]: I think the correct version is `4.6.5`?

Comment (muhammad-ammar) [reply enterprise_data/__init__.py:5]: ah...just found that we have a tag for `4.6.5` but no corresponding release.

**Review by johnnagro — APPROVED** [review]

File: enterprise_data/__init__.py (hunk 1 above)
Comment (macdiesel) [reply enterprise_data/__init__.py:5]: I added a release for 4.6.5 here: https://github.com/openedx/edx-enterprise-data/releases/tag/4.6.5

**Review by irfanuddinahmad — APPROVED** [review]
LGTM ...regarding the merge order ... I think the https://github.com/edx/prefect-flows/pull/199 PR should go last.

**Pull request merged**